Bixal/move.mil
# Feature/156848485/refine tabular data paragraph config

This pull request covers two stories in Pivotal, handling the [config](https://www.pivotaltracker.com/story/show/156848485) and [theming](https://www.pivotaltracker.com/story/show/156731753) changes to get that tabular data section looking nice and neat. I was able to iron out a couple kinks that were in those tables on the live site as an added bonus.

Feld will have to migrate the existing tabular data paragraph content into the new structure, but I did it earlier locally and it only took about 30-40 minutes so it's not too bad. Also, the switch from svgs to pngs may have an impact, but we'll deal with that separately if need be.


## Checklist

I have…

      against my changes

## Summary of Changes

This pull request…

- Creates a new "Tabular Data Wrapper" paragraph type. Its field serves as a container for unlimited instances of the original "Tabular Data" paragraph type (but no others).
- Removes the original "Tabular Data" paragraph type as an option in the main paragraphs reference field of basic pages (we can revisit this if it becomes part of the solution to the special item jumplinks question). It also removes the tutorial slide paragraph type as an option, since that one is only used on a different (Tutorial) content type.
- Adds a new twig template for the wrapper and reworks the original paragraph template to convert the table structure to css-only (far more flexible than HTML tables, which should only ever be used to present actual tables of actual data and arguably not even then) and change the table root to encompass the whole wrapper with each paragraph as a table row.
- Reworks the css to get the tables to match what's on the live site.

## Testing

To verify the changes proposed in this pull request…

1. Pull code and import config
1. Build css assets from theme folder
1. Migrate the existing tabular data paragraph content into new ones under the wrapper paragraph.
1. I don't actually expect you to migrate the paragraphs, so just check out the screenshots below. And if we need to get it verified soon on alpha before Feld can get to it, give me an admin login to alpha so I can do the migration myself once this PR is merged.

## Screenshots

### Top of Special Items table on Entitlements page:

[Live site](https://www.move.mil/entitlements#special-item-info):
<img width="818" alt="screen shot 2018-04-17 at 6 20 54 pm" src="https://user-images.githubusercontent.com/29130580/38902007-86d509de-426c-11e8-8e2b-4fcce3a15637.png">

Local:
<img width="1081" alt="screen shot 2018-04-17 at 6 19 47 pm" src="https://user-images.githubusercontent.com/29130580/38901937-37addbec-426c-11e8-904d-82aeab511f0e.png">

Local, mobile-esque screen width:
<img width="442" alt="screen shot 2018-04-17 at 6 24 37 pm" src="https://user-images.githubusercontent.com/29130580/38902033-af9a617a-426c-11e8-8f1f-b3ffc85dfadb.png">
